### ISSUE TITLE
[BUG] Launching Longhorn in two namespaces bricks both

### ISSUE THREAD

**Issue by danmanners** [longhorn/longhorn#2296]
**Describe the bug**
I accidentally re-deployed longhorn in a second namespace ('longhorn'), and it seemed to break Longhorn in my original namespace ('longhorn-system').

**To Reproduce**
1. Deploy Longhorn per the documentation
2. Deploy Longhorn again per the documentation, but in a second namespace (I used `longhorn` instead of `longhorn-system`)
3. Try to remove the second one

**Expected behavior**
I should be able to remove the second deployment

**Support Bundle**
[longhorn-support-bundle_675e1c78-abc2-48cd-944f-dab52fe5409c_2021-03-04T15-21-51Z.zip](https://github.com/longhorn/longhorn/files/6084707/longhorn-support-bundle_675e1c78-abc2-48cd-944f-dab52fe5409c_2021-03-04T15-21-51Z.zip)

**Environment:**
 - Longhorn version: 1.1.0
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: RKE v1.19.6
   - Number of management node in the cluster: 2
   - Number of worker node in the cluster: 4
 - Node config
   - OS type and version: CentOS 7
   - CPU per node: 2vCPU on controlplane and etcd nodes, 4vCPU on workers
   - Memory per node: 4GB on controlplane and etcd nodes, 12GB on workers
   - Disk type(e.g. SSD/NVMe): 10Gbps SAN connected to hypervisors; M.2 SSD's
   - Network bandwidth between the nodes: 1Gbps
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal):
 - Number of Longhorn volumes in the cluster: 16

**Additional context**

I accidentally launched Longhorn a second time when adding an argoProject because I stupidly entered the wrong namespace ('longhorn' instead of 'longhorn-system'), see (here)[https://github.com/danmanners/RKE-Learning-2/blob/master/Longhorn/longhorn.yaml]. When I did go through and remove it from the 'longhorn' namespace, I realized I was unable to remove the engineImage. Not only that, but it appears they are the exact same in both locations and I cannot remove it from either. This is also causing the side-effect of no new volumes can be created, and volumes that are unmounted cannot be re-mounted. Hopefully don't have to nuke my cluster.

```bash
➜  ~ k get engineimages.longhorn.io -n longhorn 
NAME          STATE   IMAGE                               REFCOUNT   BUILDDATE   AGE
ei-cf743f9c   ready   longhornio/longhorn-engine:v1.1.0   0          76d         12h
➜  ~ k get engineimages.longhorn.io -n longhorn-system
NAME          STATE   IMAGE                               REFCOUNT   BUILDDATE   AGE
ei-cf743f9c   ready   longhornio/longhorn-engine:v1.1.0   16         76d         45d
```

My fear is that now that I am aware of this issue, I am concerned that a malicious user could add the engineImage to their namespace and tank my persistent storage.

```yaml
➜  ~ k get engineimages.longhorn.io -n longhorn ei-cf743f9c -oyaml 
apiVersion: longhorn.io/v1beta1
kind: EngineImage
metadata:
  creationTimestamp: "2021-03-04T03:13:09Z"
  deletionGracePeriodSeconds: 0
  deletionTimestamp: "2021-03-04T03:17:56Z"
  finalizers:
  - longhorn.io
  generation: 2
  managedFields:
  - apiVersion: longhorn.io/v1beta1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:finalizers:
          .: {}
          v:"longhorn.io": {}
      f:spec: {}
      f:status: {}
    manager: longhorn-manager
    operation: Update
    time: "2021-03-04T03:13:09Z"
  name: ei-cf743f9c
  namespace: longhorn
  resourceVersion: "18209764"
  selfLink: /apis/longhorn.io/v1beta1/namespaces/longhorn/engineimages/ei-cf743f9c
  uid: 28d926b0-8f36-4138-bf4b-87a13766d110
spec:
  image: longhornio/longhorn-engine:v1.1.0
status:
  buildDate: "2020-12-18T00:38:19+00:00"
  cliAPIMinVersion: 3
  cliAPIVersion: 4
  conditions:
    ready:
      lastProbeTime: ""
      lastTransitionTime: "2021-03-04T03:13:20Z"
      message: Engine image ei-cf743f9c (longhornio/longhorn-engine:v1.1.0) become
        ready
      reason: ""
      status: "True"
      type: ready
  controllerAPIMinVersion: 3
  controllerAPIVersion: 3
  dataFormatMinVersion: 1
  dataFormatVersion: 1
  gitCommit: b9e072eb19cfcc27894e10a8e60794c66ca918f8
  noRefSince: "2021-03-04T03:13:10Z"
  ownerID: rke-worker-3
  refCount: 0
  state: ready
  version: v1.1.0
➜  ~ k get engineimages.longhorn.io -n longhorn-system ei-cf743f9c -oyaml
apiVersion: longhorn.io/v1beta1
kind: EngineImage
metadata:
  creationTimestamp: "2021-01-18T14:21:09Z"
  finalizers:
  - longhorn.io
  generation: 1
  managedFields:
  - apiVersion: longhorn.io/v1beta1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:finalizers:
          .: {}
          v:"longhorn.io": {}
      f:spec: {}
      f:status: {}
    manager: longhorn-manager
    operation: Update
    time: "2021-01-18T14:21:09Z"
  name: ei-cf743f9c
  namespace: longhorn-system
  resourceVersion: "18221024"
  selfLink: /apis/longhorn.io/v1beta1/namespaces/longhorn-system/engineimages/ei-cf743f9c
  uid: 883e20d4-af40-4fa0-b5a8-2d3cd93a7996
spec:
  image: longhornio/longhorn-engine:v1.1.0
status:
  buildDate: "2020-12-18T00:38:19+00:00"
  cliAPIMinVersion: 3
  cliAPIVersion: 4
  conditions:
    ready:
      lastProbeTime: ""
      lastTransitionTime: "2021-03-04T03:57:47Z"
      message: Engine image ei-cf743f9c (longhornio/longhorn-engine:v1.1.0) become
        ready
      reason: ""
      status: "True"
      type: ready
  controllerAPIMinVersion: 3
  controllerAPIVersion: 3
  dataFormatMinVersion: 1
  dataFormatVersion: 1
  gitCommit: b9e072eb19cfcc27894e10a8e60794c66ca918f8
  noRefSince: ""
  ownerID: rke-worker-3
  refCount: 16
  state: ready
  version: v1.1.0
```

**Comment by joshimoo** [comment]
We currently do not support running longhorn outside of the longhorn-system namespace.
It's something we are working on.

The engine images that you are showing are 2 different engine image custom resources that both have a reference to the same longhorn-engine (docker image) but operate independently of each other.

**Comment by danmanners** [comment]
Hey @joshimoo, thanks for the really fast response.

To be clear, I did not intend to launch it outside of the "longhorn-system" namespace; I made a typo and realized it a few seconds too late.

The issue that I'm running into is that I can no longer use longhorn in the "longhorn-system" namespace successfully, and attempting to remove the engineImage from the "longhorn" namespace does not work.

**Comment by danmanners** [comment]
Quick update, running the uninstall script from the docs [here](https://longhorn.io/docs/1.1.0/deploy/uninstall/#uninstalling-longhorn-using-kubectl) seems to have removed the resources, but unclear if any of my pods with persistent storage are actually coming back up.

```bash
curl -q https://raw.githubusercontent.com/longhorn/longhorn/v1.1.0/uninstall/uninstall.yaml | \
  sed -e 's/longhorn-system/longhorn/g' | \
  kubectl apply -f -
```

I'll update when I have more.

**Comment by joshimoo** [comment]
If you remove the finalizer from the engine image or any other kubernetes longhorn resource, than it will be deleted.
```
  finalizers:
  - longhorn.io
```

You can use kubectl edit to do so, have a look here:
https://longhorn.io/docs/1.1.0/deploy/uninstall/#problems-with-crds

There is a uninstall.yaml but it's only supposed to be used for the longhorn-system namespace.

**Comment by danmanners** [comment]
It would appear that after a full restart of **ALL** Longhorn containers things are working.

**TL;DR**: I turned it all off and back on.

Remediation steps:
1. Scaled all deployments/statefulsets using Longhorn volumes to zero
2. Deleted/Restarted ALL Longhorn pods
3. Waited for all pods to come back up (~3-4m)
4. Scaled deployments/statefulsets using Longhorn volumes back to their original count.

I'm not entirely surprised that restarting all the longhorn containers fixed the issue, but a bit frustrating for sure.

Closing this issue. @joshimoo thanks again for the super fast response!
